ibis/expr/types/relations.py

```diff
@@ -867,6 +867,54 @@ def fillna(
                 )
         return ops.FillNa(self, replacements).to_expr()
 
+    def unpack(self, *columns: str) -> Table:
+        """Project the struct fields of each of `columns` into `self`.
+
+        Existing fields are retained in the projection.
+
+        Parameters
+        ----------
+        columns
+            String column names to project into `self`.
+
+        Returns
+        -------
+        Table
+            The child table with struct fields of each of `columns` projected.
+
+        Examples
+        --------
+        >>> schema = dict(a="struct<b: float, c: string>", d="string")
+        >>> t = ibis.table(schema, name="t")
+        >>> t
+        UnboundTable: t
+          a struct<b: float64, c: string>
+          d string
+        >>> t.unpack("a")
+        r0 := UnboundTable: t
+          a struct<b: float64, c: string>
+          d string
+
+        Selection[r0]
+          selections:
+            b: StructField(r0.a, field='b')
+            c: StructField(r0.a, field='c')
+            d: r0.d
+
+        See Also
+        --------
+        ibis.expr.types.structs.StructValue.lift
+        """
+        columns_to_unpack = frozenset(columns)
+        result_columns = []
+        for column in self.columns:
+            if column in columns_to_unpack:
+                expr = self[column]
+                result_columns.extend(expr[field] for field in expr.names)
+            else:
+                result_columns.append(column)
+        return self[result_columns]
```

Comment:
This currently will silently no-op if a column named passed in doesn't exist in the table. That should probably error instead.

Also, does the order of columns ever matter in ibis? If so, we should probably iterate over the passed in `columns` in order.

Reply:
> This currently will silently no-op if a column named passed in doesn't exist in the table. That should probably error instead.

Makes sense.

> Also, does the order of columns ever matter in ibis? If so, we should probably iterate over the passed in columns in order.

Column order matters in ibis.

In this case I'm preserving the relative order of the field in the output. What alternative did you have in mind?

Reply:
Agreed. The way I've implemented it preserves **an** order, but of course not the only one. We could for example stick the unpacked fields at the beginning or the end of the projection but that seemed more arbitrary than preserving the unpacked fields existing starting offset.